### DISCUSSION TITLE
vscode apply all quick fix

### DISCUSSION THREAD

**Discussion by cacosub7** [clangd/clangd#1524] [Q&A]
In vscode-clangd there can give some useful code quick-fix action like -Wformat.  
![image](https://user-images.githubusercontent.com/46550268/223353576-9500aacc-eff3-4dd8-9c10-22153b1e884d.png)
![image](https://user-images.githubusercontent.com/46550268/223352776-032c526c-8ea2-4e61-ae20-fa30da997c1b.png)
 I've not found any option that can fix all same code problem so far. Obviously, it's tedious to fix same code problem one by one by hovering.
Is the `fix all` feature released?  If not, can i execute clangd in command line to approach this?

**Comment by HighCommander4** [comment]
Does selecting the whole file and running the `Auto Fix...` command in VSCode work?

**Reply by cacosub7** [reply]
No, it doesn't.
When i did so, vscode told me that
![image](https://user-images.githubusercontent.com/46550268/223369679-b672a447-4d23-4776-a092-f29f571921fc.png)

**Reply by cacosub7** [reply]
Only code actions show up. 
![image](https://user-images.githubusercontent.com/46550268/223370053-d82ef587-179d-4f45-b8ea-7beff1aa4014.png)
You still have to do it manually.

**Reply by HighCommander4** [reply]
Looks like there is a bug on the clangd side here. I filed https://github.com/clangd/clangd/issues/1536 for further investigation.

**Reply by cacosub7** [reply]
Thanks~

**Comment by cacosub7** [comment]
Further more, I noticed that the checks for clangd are composed of the clang frontend and clang-tidy. So, can warnings from the clang frontend, such as '-Wformat', be fixed from the command line like clang-tidy? Or does clang-tidy support detecting and fixing warnings from the clang frontend in VS Code?
![image](https://user-images.githubusercontent.com/46550268/224644200-9a858bd1-1e62-4a25-9a81-c32f8caafa5e.png)

**Reply by HighCommander4** [reply]
> can warnings from the clang frontend, such as '-Wformat', be fixed from the command line like clang-tidy?

Good question! It seems like they can, with the following command:

```
clang -Xclang -fixit <other_flags> <input_file>
```

Note that this command will modify the input file in-place.
